src/main/java/storagesystem/viewcontroller/inventory/DetailedItemViewController.java

```diff
@@ -0,0 +1,359 @@
+        itemPageIDLabel.setText("ID: " + id);
+    }
+
+    private void setAmountLabel(String amount) {
+        itemPageAmountTA.setText(amount);
+    }
+
+
+    private void setTeamOwnerLabel(String teamOwner) {
+        itemPageTeamOwnerLabel.setText("Owner: " + teamOwner);
+    }
+
+    private void setReservableBtn(boolean reservable) {
+        itemPageReserveBtn.setDisable(!reservable);
+    }
+
+    /**
+     * Below this line contains listener methods
+     */
+    private List<DetailedItemViewListener> detailListeners = new ArrayList<>();
+    private List<saveButtonClickedListener> saveButtonListeners = new ArrayList<>();
+
+
+    /**
+     * adds a listener to this object.
+     *
+     * @param listener
+     */
+    public void addDetailListener(DetailedItemViewListener listener) {
+        detailListeners.add(listener);
+    }
+
+    public void addSaveButtonListener(saveButtonClickedListener listener) {
+        saveButtonListeners.add(listener);
+    }
+
+/******************************************************************************************************
+ * methods below is about EDITITNG and SAVING an iReservable.
+ ******************************************************************************************************/
+
+    /**
+     * this method creates a new item and registers it to the current team.
+     *
+     * @author Pär Aronsson
+     */
+    public void editItem() {
+
+        itemPageReserveBtn.setVisible(false);
+        itemPageAmountTA.setEditable(true);
+        itemPageSaveButton.setVisible(true);
+        itemPageDescriptionTA.setEditable(true);
+        itemPageUserRequirementsTA.setEditable(true);
+        itemPageNameTA.setEditable(true);
+        isReservableChoiceBox.setDisable(false);
+        editPane.toFront();
+        itemPageConditionSlider.setDisable(false);
+        itemPageImageView.setOnMouseClicked(event -> {
+            changeItemImage();
+        });
+    }
+
+    /**
+     * saves the changes to the current item.
+     */
+    public void saveItem() {
+
+        reservableItem.setName(itemPageNameTA.getText());
+        reservableItem.setDescription(itemPageDescriptionTA.getText());
+        reservableItem.setUserRequirements(itemPageUserRequirementsTA.getText());
+        reservableItem.setImage(itemPageImageView.getImage());
+        reservableItem.setReservable(isReservableChoiceBox.getSelectionModel().getSelectedIndex() == 0);
+        saveCondition((int) itemPageConditionSlider.getValue());
+        saveLocation(itemPageLocationChoicebox.getValue().toString());
+        reservableItem.setAmount(Integer.valueOf(itemPageAmountTA.getText()));
+        for (saveButtonClickedListener l : saveButtonListeners) {
+            l.saveButtonClicked();
+        }
+
+    }
+
+    /**
+     * checks if the choicebox string value is equal to the location name and if so, it saves the location to the item.
+     *
+     * @param s
+     */
+    private void saveLocation(String s) {
```

Comment:
`String s` not very descriptive.
```suggestion
    private void saveLocation(String locationName) {
```